ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -0,0 +1,337 @@
+            assertEquals(truncatedJob.getMasterObjectListResult().getOriginalSizeInBytes(), Files.size(objPath1));
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void cancelJobWithForce() throws IOException, SignatureException, XmlProcessingException, URISyntaxException, InterruptedException {
+
+        final int testTimeOutSeconds = 5;
+        final String bucketName = "test_force_cancel_job";
+        final String book1 = "beowulf.txt";
+        final Path objPath1 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book1);
+        final Ds3Object obj1 = new Ds3Object(book1, Files.size(objPath1));
+        final Ds3Object obj2 = new Ds3Object("place_holder", 5000000);
+
+        try {
+            client.putBucket(new PutBucketRequest(bucketName));
+            Ds3ClientHelpers.wrap(client).ensureBucketExists(bucketName);
+
+            final Ds3ClientHelpers.Job putJob = Ds3ClientHelpers.wrap(client).startWriteJob(bucketName, Lists.newArrayList(obj1, obj2));
+            final UUID jobId = putJob.getJobId();
+            final SeekableByteChannel book1Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book1);
+            client.putObject(new PutObjectRequest(bucketName, book1, book1Channel, jobId, 0, Files.size(objPath1)));
+
+            //make sure black pearl has updated it's job to show 1 object in cache
+            final long startTimePutObject = System.nanoTime();
+            long cachedSize =0;
+            while (cachedSize == 0) {
+                Thread.sleep(500);
+                final MasterObjectList mol = client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult();
+                cachedSize = mol.getCachedSizeInBytes();
+                assertTrue((System.nanoTime() - startTimePutObject)/1000000000 < testTimeOutSeconds );
+            }
+
+            final CancelJobSpectraS3Response responseWithForce = client
+                    .cancelJobSpectraS3(new CancelJobSpectraS3Request(jobId).withForce(true));
+            assertEquals(responseWithForce.getStatusCode(), 204);
+
+            //Allow for lag time before canceled job appears~1.5 seconds in unloaded system
+            final long startTimeCanceledUpdate = System.nanoTime();
+            boolean isACanceledJob = false;
+            while (!isACanceledJob) {
+                Thread.sleep(500);
+                GetCanceledJobsSpectraS3Response canceledJobs = client.getCanceledJobsSpectraS3(new GetCanceledJobsSpectraS3Request());
+                for (CanceledJob canceledJob : canceledJobs.getCanceledJobListResult().getCanceledJobs()){
+                    if (canceledJob.getId().equals(jobId)){
+                        isACanceledJob = true;
+                    }
+                }
+                assertTrue((System.nanoTime() - startTimeCanceledUpdate)/1000000000 < testTimeOutSeconds );
```

Comment:
There are some asserts to handle this instead, but I believe you have to include 'org.hamcrest:hamcrest-library:1.3' as a testCompile dependency and then you can use `assertThat((System.nanoTime() - startTimeCanceledUpdate)/1000000000, is(lessThan(testTimeOutSeconds)))`